### PR TITLE
tweaks to rapidpro side for current twitter activity APIs

### PR DESCRIPTION
**Pull request by nicpottier**
Tweaks to current twitter activity to support current "beta" API endpoints.

### CODE REVIEW

File: temba/channels/types/twitter_activity/type.py [hunk 1]
```diff
@@ -32,31 +30,28 @@ class TwitterActivityType(ChannelType):
...
+            client.subscribe_to_webhook(config['env_name'])
+        except Exception as e:
+            print(e)
 
     def deactivate(self, channel):
         config = channel.config_json()
         client = TembaTwython(config['api_key'], config['api_secret'], config['access_token'], config['access_token_secret'])
-
-        client.delete_webhook(config['webhook_id'])
+        client.delete_webhook(config['env_name'])
```
Comment (nicpottier) [inline temba/channels/types/twitter_activity/type.py]: Twitter claims only one webhook is supported.

File: temba/channels/types/twitter_activity/views.py [hunk 2]
```diff
@@ -15,6 +18,7 @@ class Form(ClaimViewMixin.Form):
         api_secret = forms.CharField(label=_('Consumer Secret'))
         access_token = forms.CharField(label=_('Access Token'))
         access_token_secret = forms.CharField(label=_('Access Token Secret'))
+        env_name = forms.CharField(label=_('Environment Name'))
```
Comment (rowanseymour) [inline temba/channels/types/twitter_activity/views.py]: What is this?

Comment (nicpottier) [reply temba/channels/types/twitter_activity/views.py]: There's an "environment" your account is associated with which is then used in the API calls. Guessing that is just a beta thing.

Comment (rowanseymour) [reply temba/channels/types/twitter_activity/views.py]: ok weird - that didn't exist when I tried this stuff previously

File: temba/channels/types/twitter_activity/type.py [hunk 3]
```diff
@@ -32,31 +30,28 @@ class TwitterActivityType(ChannelType):
     show_config_page = False
     free_sending = True
 
+    def is_available_to(self, user):
+        return user.is_beta()
```
Comment (rowanseymour) [inline temba/channels/types/twitter_activity/type.py]: You mean to keep this still beta gated? Alternatively we need to explain in the blurb that this is beta on the Twitter side

File: temba/channels/types/twitter_activity/type.py [hunk 4]
```diff
@@ -32,31 +30,28 @@ class TwitterActivityType(ChannelType):
...
-
-        # save this webhook for later so we can delete it
-        config['webhook_id'] = webhook['id']
-        channel.config = json.dumps(config)
-        channel.save(update_fields=('config',))
+        try:
+            client.register_webhook(config['env_name'], callback_url)
+            client.subscribe_to_webhook(config['env_name'])
+        except Exception as e:
+            print(e)
```
Comment (rowanseymour) [inline temba/channels/types/twitter_activity/type.py]: We want this in Sentry?

File: temba/channels/types/twitter_activity/type.py (hunk 3 above)
Comment (nicpottier) [reply temba/channels/types/twitter_activity/type.py]: Ya, still going to be beta-gated.

**Pull request merged**